src/EFCore.MySql/Scaffolding/Internal/MySqlDatabaseModelFactory.cs

```diff
@@ -509,6 +509,12 @@ private void GetConstraints(DbConnection connection, IReadOnlyList<DatabaseTable
                         {
                             var referencedTableName = reader.GetString(2);
                             var referencedTable = tables.FirstOrDefault(t => t.Name == referencedTableName);
+                            if (referencedTable == null)
+                            {
+                                // if the casing of the constraint doesn't match the casing of the table name, try and find a matching table with a
+                                // different case, but ensure that there's only one possible match or we can't be certain which is the right table.
+                                referencedTable = tables.SingleOrDefault(t => string.Equals(t.Name, referencedTableName, StringComparison.OrdinalIgnoreCase));
```

Comment:
On the case *in*sensitive branch, we do not want any ambiguity between table names. The easiest way to archive this is to throw in case there are multiple tables that would satisfy the comparison (which should never happen).
So change `SingleOrDefault()` to `Single()`.

Reply:
SingleOrDefault has been changed to Single